Content.Server/Botany/Components/ThornyImmuneComponent.cs

```diff
@@ -0,0 +1,7 @@
+namespace Content.Server.Botany.Components;
+[RegisterComponent]
+public sealed partial class ThornyImmuneComponent : Component
+{
+    [DataField]
+    public bool ThornImmune = true;
```

Comment:
`ThornyImmuneComponent` seems that it should automatically confer thron resistance, so this data field is not necessary.